contracts/facets/Token/TokenBridgeInternal.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.8.19;
+
+import { TokenStorage as Storage } from "./Storage.sol";
+import { TokenInternal } from "./TokenInternal.sol";
+
+import { ITokenBridgeInternal } from "./ITokenBridgeInternal.sol";
+
+import { AxelarExecutable } from "@axelar/executable/AxelarExecutable.sol";
+
+import { IAxelarGasService } from "@axelar/interfaces/IAxelarGasService.sol";
+
+import { IAxelarGateway } from "@axelar/interfaces/IAxelarGateway.sol";
+
+/// @title TokenBridgeInternal
+/// @notice it contains internal functions of the TokenBridgeInternal contract
+abstract contract TokenBridgeInternal is
+    TokenInternal,
+    AxelarExecutable,
+    ITokenBridgeInternal
+{
+    /// @notice Minimum gas required to execute a transaction through Axelar Gateway
+    uint256 public constant MIN_GAS_PER_TX = 0.001 ether;
+
+    /// @notice Axelar Gas Service contract in charge of handling gas disposal on other chains
+    IAxelarGasService public immutable axelarGasService;
+    constructor(address gateway, address gasService) AxelarExecutable(gateway) {
+        if (gasService == address(0)) revert TokenBridge__InvalidAddress();
+        axelarGasService = IAxelarGasService(gasService);
+    }
+
+    /// @notice it check if the destination chain is a valid input
+    /// @param destinationChain the destination chain
+    /// @return true if the destination chain is valid
+    function _isDestinationChainValid(
+        string calldata destinationChain
+    ) internal pure returns (bool) {
+        return bytes(destinationChain).length != 0;
+    }
+
+    /// @notice it check if the destination address is a valid input
+    /// @param destinationAddress the destination address
+    /// @return true if the destination address is valid
+    function _isDestinationAddressValid(
+        string calldata destinationAddress
+    ) internal pure returns (bool) {
+        return bytes(destinationAddress).length != 0;
```

Comment:
do we want to add checks for EVM, Solana and Polkadot addresses?

Reply:
To make the system ad flexible as possible I'll add a variable uint256 addressLengtsBitMap, I'll add setters for enabling or disabling a certain length of addresses. This will ensure maximum efficiency and flexibility in enabling or disabling certain chains